### PR TITLE
Update a-star-search.md

### LINKED ISSUE #3596: [Edit] Subject: A* search
### Type of Edit (select all that apply)

Clarify unclear content and/or terms, Add new content (definitions, codeblocks, etc.)

### Description (optional)

Some properties of A* search algorithm are missing like optimality and the completeness.Also some terms are unclear.

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by Sahilkumar19**
Description
Fixes #3596 
This PR enhancing the clarity and readability in the topic of A* search algorithm.Added some important properties like optimality and colpleteness.
closes #3596 

Type of changes
- Adding a new entry
- Editing an existing entry (fixing a typo, bug, issues, etc)
- Updating the documentation

### CODE REVIEW

**Review by Sahilkumar19 — COMMENTED** [review]
Reviewed the changes.

File: content/ai/concepts/search-algorithms/terms/a-star-search/a-star-search.md [hunk 1]
```diff
@@ -12,17 +12,24 @@ CatalogContent:
   - 'paths/machine-learning-ai-engineering-foundations'
 ---
 
-**A\* Search** is an informed [best-first search algorithm](https://www.codecademy.com/resources/docs/ai/search-algorithms/best-first-search) that efficiently determines the lowest cost path between any two nodes in a directed weighted graph with non-negative edge weights. This algorithm is a variant of Dijkstra's algorithm. A slight difference arises from the fact that an evaluation function is used to determine which node to explore next.
+**A\* Search** is an informed [best-first search algorithm](https://www.codecademy.com/resources/docs/ai/search-algorithms/best-first-search) It evaluates the nodes by combining **g(n)** the exact cost to reach the node n from the starting point, and **h(n)** the (heuristic) approximate cost to get the 'n' to the goal.
```
Comment (SSwiniarski) [inline content/ai/concepts/search-algorithms/terms/a-star-search/a-star-search.md:15]:
```suggestion
**A\* Search** is an informed [best-first search algorithm](https://www.codecademy.com/resources/docs/ai/search-algorithms/best-first-search) It evaluates the nodes by combining **g(n)** the exact cost to reach the node n from the starting point, and **h(n)** the (heuristic) approximate cost to get from 'n' to the goal.
```

File: content/ai/concepts/search-algorithms/terms/a-star-search/a-star-search.md [hunk 2]
```diff
@@ -12,17 +12,24 @@ CatalogContent:
...
+
+**f(n)** = **h(n)** + **g(n)**
+**g(n)** = The exact cost to reach node n from the starting node.
+**h(n)** = The heuristic cost of the cheapest path from node n to the goal.
+**f(n)** = Estimated cost of the cheapest solution from the starting point to the goal state through the goal state.
```
Comment (SSwiniarski) [inline content/ai/concepts/search-algorithms/terms/a-star-search/a-star-search.md:20]:
```suggestion
- **f(n)** = **h(n)** + **g(n)**
- **g(n)** = The exact cost to reach node n from the starting node.
- **h(n)** = The heuristic cost of the cheapest path from node n to the goal.
- **f(n)** = Estimated cost of the cheapest solution from the starting point to the goal through n.
```

File: content/ai/concepts/search-algorithms/terms/a-star-search/a-star-search.md [hunk 3]
```diff
@@ -12,17 +12,24 @@ CatalogContent:
...
+
+Provided that heuristic function should satisfy certain conditions A* search is both complete and optimal given that **h(n)** should be admissible heuristic means that it should never overestimate the cost to reach the goal and **h(n)** must be consistent, i.e., it satisfies the triangle inequality.
```
Comment (SSwiniarski) [inline content/ai/concepts/search-algorithms/terms/a-star-search/a-star-search.md:22]:
```suggestion
A* search is both complete and optimal given that **h(n)** should be **admissible**, meaning that it should never overestimate the cost to reach the goal, and **h(n)** must be **consistent**, i.e., it satisfies the triangle inequality.
```

File: content/ai/concepts/search-algorithms/terms/a-star-search/a-star-search.md [hunk 4]
```diff
@@ -12,17 +12,24 @@ CatalogContent:
...
 
 ```pseudo
-f(x) = g(x) + h(x)
+f(n) = g(n) + h(n)
 ```
 
-Where **g(x)** represents the cost to get to node **x** and **h(x)** represents the estimated cost to arrive at the goal node from node **x**.
+Where **g(n)** represents the cost to get to node **n** and **h(n)** represents the estimated cost to arrive at the goal node from node **n**.
 
 For the algorithm to generate the correct result, the evaluation function must be **admissible**, meaning that it never overestimates the cost to arrive at the goal node.
```
Comment (SSwiniarski) [inline content/ai/concepts/search-algorithms/terms/a-star-search/a-star-search.md:34]:
```suggestion
```
This line is now redundant.

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@Sahilkumar19 I did a first review and left a few comments.

File: content/ai/concepts/search-algorithms/terms/a-star-search/a-star-search.md [hunk 5]
```diff
@@ -12,19 +12,25 @@ CatalogContent:
   - 'paths/machine-learning-ai-engineering-foundations'
 ---
 
-**A\* Search** is an informed [best-first search algorithm](https://www.codecademy.com/resources/docs/ai/search-algorithms/best-first-search) that efficiently determines the lowest cost path between any two nodes in a directed weighted graph with non-negative edge weights. This algorithm is a variant of Dijkstra's algorithm. A slight difference arises from the fact that an evaluation function is used to determine which node to explore next.
+**A\* Search** is an informed [best-first search algorithm](https://www.codecademy.com/resources/docs/ai/search-algorithms/best-first-search) It evaluates the nodes by combining **g(n)** the exact cost to reach the node n from the starting point, and **h(n)** the (heuristic) approximate cost to get from 'n' to the goal.
+
+**f(n)** = **h(n)** + **g(n)**
+**g(n)** = The exact cost to reach node n from the starting node.
+**h(n)** = The heuristic cost of the cheapest path from node n to the goal.
+**f(n)** = Estimated cost of the cheapest solution from the starting point to the goal through n.
```
Comment (SSwiniarski) [inline content/ai/concepts/search-algorithms/terms/a-star-search/a-star-search.md:20]:
```suggestion
- **f(n)** = **h(n)** + **g(n)**
- **g(n)** = The exact cost to reach node n from the starting node.
- **h(n)** = The heuristic cost of the cheapest path from node n to the goal.
- **f(n)** = Estimated cost of the cheapest solution from the starting point to the goal through n.
```
This needs to be a bulleted list.

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@Sahilkumar19 one more comment.

File: content/ai/concepts/search-algorithms/terms/a-star-search/a-star-search.md (hunk 5 above)
Comment (Sahilkumar19) [reply content/ai/concepts/search-algorithms/terms/a-star-search/a-star-search.md]: I have made the suggested changes.

**Review by SSwiniarski — APPROVED** [review]
@Sahilkumar19 Looks good for a second review.

**Comment by SSwiniarski** [comment]
> > @Sahilkumar19 Looks good for a second review.
> 
> Who will review it for the second time (just asking)

@Sahilkumar19 one of the other maintainers. It's not set exactly which one, that depends on everyone's workload.

**Comment by caupolicandiaz** [comment]
@Sahilkumar19 Apologies for the delay moving forward with this PR. We've reconsidered these edits and decided not to incorporate them. We hope you can find another issue or concept to work on and contribute to docs. Thanks

**Pull request closed without merging**